README.rst

```diff
@@ -1,95 +1,167 @@
-=========
+#########
 I Promise
-=========
+#########
+
 .. image:: https://badge.fury.io/py/ipromise.svg
     :target: https://badge.fury.io/py/ipromise
 
-This repository provides a Python base class, and various decorators for specifying promises relating to inheritance.
-It provides three inheritance patterns:
+This repository, ``ipromise``, provides a Python base class and
+decorators to guarantee inheritance "promises".
+
+Three inheritance patterns are provided:
+
+* implementing
+* overriding
+* augmenting
+
+Using the ``ipromise`` inheritance patterns can ensure an inheritance hierarchy
+is used as intended by forcing a build-time or run-time failure when not.
+
+----
+
+.. contents::
+
+----
+
+**********
+Installing
+**********
 
-* implementing,
-* overriding, and
-* augmenting.
+.. code-block:: shell
 
-Base class
-==========
-Checking promises depends on inheritance from the base class ``AbstractBaseClass``.  Unlike the standard library's similar class ``abc.ABCMeta``, ``AbstractBaseClass`` does not bring in any metaclasses.  This is thanks to Python 3.6's PEP 487, which added ``__init_subclass__``.
+    python -m pip install ipromise
+
+***
+Use
+***
+
+Base class ``AbstractBaseClass``
+================================
+
+Checking promises depends on inheritance from the base class
+``AbstractBaseClass``.  Unlike the standard library's similar class
+``abc.ABCMeta``, the ``AbstractBaseClass`` does not bring in any metaclasses.
+This is thanks to Python 3.6 `PEP 487
+<https://peps.python.org/pep-0487/>`_
+which added ``__init_subclass__``.
+
+Implementing with ``implements``
+================================
+
+*Implementing* is the pattern whereby an inheriting class's method implements an
+abstract method from a base class method.
 
-Implementing
-============
-*Implementing* is the pattern whereby an inheriting class's method implements an abstract method from a base class method.
 It is declared using the decorators:
 
-* ``abc.abstractmethod`` from the standard library, and
-* ``implements``, which indicates that a method implements an abstract method in a base class
+* ``abc.abstractmethod`` from the standard library
+* ``implements`` which indicates that a method implements an abstract method in
```

Comment:
Don't lose the comma before "which".  It's needed when "which" is used to explain rather than narrow.  "which" without a comma is equivalent to "that" and is used in narrowing phrases.

Don't lose the ", and".  This is a still a senence.  Also there's a period missing in the original text that could be added here.